### PR TITLE
Bump macOS deployment target for TSan to 10.13 for LLVM only preset

### PR DESCRIPTION
**Pull request by edymtt**
This allows LLVM TSAN tests to pass when using Xcode 14.3.

Long term we want to switch to `LLVM_USE_RUNTIMES` to build compiler-rt, so we can leverage the value of `SANITIZER_MIN_OSX_VERSION` set in `cmake_product.py`.
In the meantime, we prefer enforcing the deployment in the only preset that needs this, since this currently requires leveraging an intermediate variable (`CLANG_COMPILER_RT_CMAKE_ARGS`) which cannot be easily augmented in presets mixins.

Addresses rdar://106441106

### CODE REVIEW

**Comment by edymtt** [comment]
@swift-ci please test

**Review by compnerd — APPROVED** [review]

**Review by yln — APPROVED** [review]
Thanks!

I've asked @thetruestblue to run a local llvm build with `-DSANITIZER_MIN_OSX_VERSION:STRING=10.13` for a quick sanity check.  Please wait for her approval.

**Comment by yln** [comment]
> Thanks!
> 
> I've asked @thetruestblue to run a local llvm build with `-DSANITIZER_MIN_OSX_VERSION:STRING=10.13` to verify this. Please wait for here approval.

**Comment by yln** [comment]
Apologies for my fat fingers 😒

**Comment by edymtt** [comment]
@swift-ci please test Linux

**Comment by thetruestblue** [comment]
I have locally tested Sanitizers at the higher min version with no issues. Keep in mind that currently the sanitizer's min version default is 10.10, and let us know if you see any issues with this bump.

**Review by thetruestblue — APPROVED** [review]

**Review by drexin — APPROVED** [review]

**Comment by etcwilde** [comment]
preset=buildbot_incremental,tools=RA,llvm-only
@swift-ci Please test with preset macOS Platform

**Comment by edymtt** [comment]
@swift-ci please test macOS

**Comment by edymtt** [comment]
@swift-ci please build toolchain

**Comment by edymtt** [comment]
@swift-ci please test macOS

**Pull request merged**
